compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4

```diff
@@ -156,6 +157,15 @@ constantDefinition
     :   PUBLIC? CONST typeName? Identifier ASSIGN constantExpression SEMICOLON
     ;
 
+enumDefinition
+    :   documentationString? annotationAttachment* PUBLIC? ENUM Identifier LEFT_BRACE
```

Comment:
The spec says the attachment point for `module-enum-decl` is `type`.

But this seems to be matching for consts now.

```ballerina
public annotation X on type;
public const annotation Y on source const;

@X // should work, doesn't
@Y // should fail, doesn't
public enum Color {
    RED,
    BLUE
}
```

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/23308 tracks fixing this,